ambry-cloud/src/main/java/com.github.ambry.cloud/VcrRequests.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.cloud;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.clustermap.DataNodeId;
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.commons.ServerMetrics;
+import com.github.ambry.network.Request;
+import com.github.ambry.network.RequestResponseChannel;
+import com.github.ambry.notification.NotificationSystem;
+import com.github.ambry.protocol.RequestOrResponseType;
+import com.github.ambry.replication.FindTokenHelper;
+import com.github.ambry.replication.ReplicationEngine;
+import com.github.ambry.server.AmbryRequests;
+import com.github.ambry.server.ServerErrorCode;
+import com.github.ambry.server.StoreManager;
+import com.github.ambry.store.Store;
+import com.github.ambry.store.StoreKeyConverterFactory;
+import com.github.ambry.store.StoreKeyFactory;
+import java.io.IOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Request implementation class for Vcr. All requests to the vcr server are
+ * handled by this class.
+ */
+public class VcrRequests extends AmbryRequests {
```

Comment:
Why extends `AmbryRequests` instead of implementing `RequestAPI`?
`VcrRequests` logic can be independent from  `AmbryRequests`, right?

Reply:
This is because except for a few helper utils that depend upon the idiosyncrasies of cloud, ambryrequests will be same as vcrrequests. (as discussed offline).